mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala

```diff
@@ -179,7 +179,7 @@ abstract class GeneralizedLinearAlgorithm[M <: GeneralizedLinearModel]
    */
   def run(input: RDD[LabeledPoint]): M = {
     if (numFeatures < 0) {
-      numFeatures = input.map(_.features.size).first()
+      numFeatures = input.first().features.size
```

Comment:
I see, I had thought,  `input.map(_.features.size)` is evaluated first and then from that the first element is extracted. Should I revert this?